Discouraged file produced by `--discouraged` does not pass verifiers

Being able to update the discouraged file automatically is a very nice feature, unfortunately it is currently not usable due to incorrect output generation.

I created a new branch in my set.mm fork called [discouraged-test](https://github.com/GinoGiotto/set.mm/tree/discouraged-test). I updated that branch with the most recent version of set.mm, then I executed:

`metamath-knife set.mm --discouraged discouraged`

After committing the newly produced discouraged file, these errors appear:

https://github.com/GinoGiotto/set.mm/actions/runs/7398403197/job/20127485347

<pre>
Checking discouraged file:
--- discouraged	2024-01-03 14:19:14.827913367 +0000
+++ discouraged.new	2024-01-03 14:19:58.135999643 +0000
@@ -17831,0 +17832 @@
+Proof modification of "19.21a3con13vVD" is discouraged (107 steps).
@@ -17869,0 +17871 @@
+Proof modification of "3impexpbicomiVD" is discouraged (28 steps).
@@ -18260,0 +18263 @@
+Proof modification of "con3ALTVD" is discouraged (51 steps).
@@ -18261,0 +18265 @@
+Proof modification of "con5VD" is discouraged (47 steps).
@@ -18266,0 +18271,4 @@
+Proof modification of "csbeq2gVD" is discouraged (61 steps).
+Proof modification of "csbfv12gALTVD" is discouraged (322 steps).
+Proof modification of "csbima12gALTVD" is discouraged (148 steps).
+Proof modification of "csbingVD" is discouraged (256 steps).
@@ -18267,0 +18276,5 @@
+Proof modification of "csbresgVD" is discouraged (195 steps).
+Proof modification of "csbrngVD" is discouraged (266 steps).
+Proof modification of "csbsngVD" is discouraged (204 steps).
+Proof modification of "csbunigVD" is discouraged (302 steps).
+Proof modification of "csbxpgVD" is discouraged (538 steps).
@@ -18395,0 +18409 @@
+Proof modification of "e2ebindVD" is discouraged (109 steps).
@@ -18458,0 +18473 @@
+Proof modification of "ee33VD" is discouraged (68 steps).
@@ -18527,0 +18543 @@
+Proof modification of "equncomiVD" is discouraged (19 steps).
@@ -18572,0 +18589 @@
+Proof modification of "exbiriVD" is discouraged (70 steps).
@@ -18799,0 +18817 @@
+Proof modification of "hbra2VD" is discouraged (26 steps).
@@ -18809,0 +18828 @@
+Proof modification of "idiVD" is discouraged (6 steps).
@@ -18823,0 +18843 @@
+Proof modification of "imbi13VD" is discouraged (67 steps).
@@ -19023,0 +19044 @@
+Proof modification of "notnotrALTVD" is discouraged (34 steps).
@@ -19030,0 +19052 @@
+Proof modification of "onfrALTVD" is discouraged (152 steps).
@@ -19031,0 +19054 @@
+Proof modification of "onfrALTlem1VD" is discouraged (105 steps).
@@ -19032,0 +19056 @@
+Proof modification of "onfrALTlem2VD" is discouraged (377 steps).
@@ -19033,0 +19058 @@
+Proof modification of "onfrALTlem3VD" is discouraged (218 steps).
@@ -19034,0 +19060 @@
+Proof modification of "onfrALTlem4VD" is discouraged (116 steps).
@@ -19035,0 +19062 @@
+Proof modification of "onfrALTlem5VD" is discouraged (320 steps).
@@ -19046,0 +19074 @@
+Proof modification of "ordelordALTVD" is discouraged (202 steps).
@@ -19131,0 +19160 @@
+Proof modification of "relopabVD" is discouraged (354 steps).
@@ -19173,0 +19203 @@
+Proof modification of "rspsbc2VD" is discouraged (90 steps).
@@ -19194,0 +19225 @@
+Proof modification of "sb5ALTVD" is discouraged (110 steps).
@@ -19216,0 +19248 @@
+Proof modification of "sbc3orgVD" is discouraged (180 steps).
@@ -19220,0 +19253 @@
+Proof modification of "sbcbiVD" is discouraged (59 steps).
@@ -19223,0 +19257 @@
+Proof modification of "sbcim2gVD" is discouraged (139 steps).
@@ -19226,0 +19261 @@
+Proof modification of "sbcoreleleqVD" is discouraged (176 steps).
@@ -19227,0 +19263 @@
+Proof modification of "sbcssgVD" is discouraged (229 steps).
@@ -19266,0 +19303,2 @@
+Proof modification of "simplbi2VD" is discouraged (24 steps).
+Proof modification of "simplbi2comtVD" is discouraged (42 steps).
@@ -19302,0 +19341 @@
+Proof modification of "ssralv2VD" is discouraged (147 steps).
@@ -19307,0 +19347,2 @@
+Proof modification of "sucidALTVD" is discouraged (28 steps).
+Proof modification of "sucidVD" is discouraged (24 steps).
@@ -19341,0 +19383 @@
+Proof modification of "tratrbVD" is discouraged (395 steps).
@@ -19344,0 +19387 @@
+Proof modification of "trintALTVD" is discouraged (211 steps).
@@ -19345,0 +19389 @@
+Proof modification of "trsbcVD" is discouraged (396 steps).
@@ -19350,0 +19395 @@
+Proof modification of "truniALTVD" is discouraged (220 steps).
@@ -19357,0 +19403 @@
+Proof modification of "undif3VD" is discouraged (295 steps).
@@ -19415,0 +19462 @@
+Proof modification of "vk15.4jVD" is discouraged (268 steps).
Error: Process completed with exit code 1.
</pre>

Basically metamath-knife incorrectly removes some lines about proof modification. I observed that these theorems contain an HTML table in their comments, which might be related to why the `(Proof modification is discouraged.)` tag isn't being read for them.

--- a/metamath-rs/src/comment_parser.rs
+++ b/metamath-rs/src/comment_parser.rs
@@ -475,8 +475,8 @@ impl Discouragements {
         static MODIFICATION: OnceLock<RegexSet> = OnceLock::new();
         let modification = MODIFICATION.get_or_init(|| {
             RegexSet::new([
-                r"\(Proof[ \n]+modification[ \n]+is[ \n]+discouraged\.\)",
-                r"\(New[ \n]+usage[ \n]+is[ \n]+discouraged\.\)",
+                r"\(Proof[ \r\n]+modification[ \r\n]+is[ \r\n]+discouraged\.\)",
+                r"\(New[ \r\n]+usage[ \r\n]+is[ \r\n]+discouraged\.\)",
             ])
             .unwrap()
         });

--- a/metamath-rs/src/comment_parser_tests.rs
+++ b/metamath-rs/src/comment_parser_tests.rs
@@ -326,6 +326,16 @@ fn test_discouragements() {
             usage_discouraged: true,
         }
     );
+    assert_eq!(
+        Discouragements::new(
+            b"A false axiom, experimental.  (Proof modification\r\n\
+            is discouraged.)  (New usage is discouraged.)"
+        ),
+        Discouragements {
+            modification_discouraged: true,
+            usage_discouraged: true,
+        }
+    );
 }
 
 #[track_caller]

